src/db_iterator.rs

```diff
@@ -0,0 +1,487 @@
+                self.inner,
+                key.as_ptr() as *const c_char,
+                key.len() as size_t,
+            );
+        }
+    }
+
+    /// Seeks to the next key.
+    ///
+    /// Returns true if the iterator is valid after this operation.
+    pub fn next(&mut self) {
+        unsafe {
+            ffi::rocksdb_iter_next(self.inner);
+        }
+    }
+
+    /// Seeks to the previous key.
+    ///
+    /// Returns true if the iterator is valid after this operation.
+    pub fn prev(&mut self) {
+        unsafe {
+            ffi::rocksdb_iter_prev(self.inner);
+        }
+    }
+
+    /// Returns a slice to the internal buffer storing the current key.
+    ///
+    /// This may be slightly more performant to use than the standard ``.key()`` method
+    /// as it does not copy the key. However, you must be careful to not use the buffer
+    /// if the iterator's seek position is ever moved by any of the seek commands or the
+    /// ``.next()`` and ``.previous()`` methods as the underlying buffer may be reused
+    /// for something else or freed entirely.
+    pub unsafe fn key_inner(&self) -> Option<&[u8]> {
+        if self.valid() {
+            let mut key_len: size_t = 0;
+            let key_len_ptr: *mut size_t = &mut key_len;
+            let key_ptr = ffi::rocksdb_iter_key(self.inner, key_len_ptr) as *const c_uchar;
+
+            Some(slice::from_raw_parts(key_ptr, key_len as usize))
+        } else {
+            None
+        }
+    }
+
+    /// Returns a copy of the current key.
+    pub fn key(&self) -> Option<Vec<u8>> {
+        unsafe { self.key_inner().map(|key| key.to_vec()) }
+    }
+
+    /// Returns a slice to the internal buffer storing the current value.
+    ///
+    /// This may be slightly more performant to use than the standard ``.value()`` method
+    /// as it does not copy the value. However, you must be careful to not use the buffer
+    /// if the iterator's seek position is ever moved by any of the seek commands or the
+    /// ``.next()`` and ``.previous()`` methods as the underlying buffer may be reused
+    /// for something else or freed entirely.
+    pub unsafe fn value_inner(&self) -> Option<&[u8]> {
+        if self.valid() {
+            let mut val_len: size_t = 0;
+            let val_len_ptr: *mut size_t = &mut val_len;
+            let val_ptr = ffi::rocksdb_iter_value(self.inner, val_len_ptr) as *const c_uchar;
+
+            Some(slice::from_raw_parts(val_ptr, val_len as usize))
+        } else {
+            None
+        }
+    }
+
+    /// Returns a copy of the current value.
+    pub fn value(&self) -> Option<Vec<u8>> {
+        unsafe { self.value_inner().map(|value| value.to_vec()) }
+    }
+}
+
+impl<'a> Drop for DBRawIterator<'a> {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::rocksdb_iter_destroy(self.inner);
+        }
+    }
+}
+
+impl<'a> DBIterator<'a> {
+    pub(crate) fn new(db: &DB, readopts: &ReadOptions, mode: IteratorMode) -> DBIterator<'a> {
```

Comment:
Description for pub methods ?

Reply:
Sure. I'll be making a documentation pass just before I transition this PR out of draft status.